### PR TITLE
Add search history in NVDA Find dialog box

### PR DESCRIPTION
**Pull request by ghost**
Replace the TextCtrl on a ComboBox, with the ability to specify the previous search strings.
I think it will be helpful, to not typing that already searched.

### CODE REVIEW

**Comment by michaelDCurran** [comment]
@feerrenrut: Does the editable combo box look okay visually? I don't think we've used one in the UI before.

**Comment by feerrenrut** [comment]
It doesn't look any worse. I think I must have missed the find dialog when fixing alignment and spacing. We will need to add a new case to the associateControls method in the guiHelper to handle the editable combo box.

**Comment by josephsl** [comment]
Hi,

A few things:

* Is there an issue attached to this PR?
* I think it'd be best to use a discrete branch instead of using master branch, as it allows reviewers and testers to track branches and diffs easily.

Thanks.

File: source/cursorManager.py [hunk 1]
```diff
@@ -149,6 +149,11 @@ def doFindText(self,text,reverse=False,caseSensitive=False):
 			wx.CallAfter(gui.messageBox,_('text "%s" not found')%text,_("Find Error"),wx.OK|wx.ICON_ERROR)
 		CursorManager._lastFindText=text
 		CursorManager._lastCaseSensitivity=caseSensitive
+		try:
+			CursorManager._searchHistory.remove(text)
+		except ValueError:
+			pass
+		CursorManager._searchHistory.insert(0, text)
```
Comment (josephsl) [inline source/cursorManager.py]: Hi,

Consider the following scenario: user goes through several rounds of searching with different texts, and now the user types a previously searched term somehow. What would you do if this happens - just prepend whatever the text is, or see if the term is already part of search history?

Thanks.

Comment (josephsl) [inline source/cursorManager.py]: Hi,

Consider the following scenario: user goes through several rounds of searching with different texts, and now the user types a previously searched term somehow. What would you do if this happens - just prepend whatever the text is, or see if the term is already part of search history?

Thanks.

**Comment by josephsl** [comment]
Hi,

After looking at the diff, several things come to my mind:

* What would you do if users enter a previously searched term - prepend or check?
* Wouldn't it be better if only the search history is consulted instead of storing the last searched term in the cursor manager? That way the text to be searched could be looked up from search history (top item) and allows NVDA+F3/NVDA+Shift+F3 logic to check if the search history is empty or not.

Thanks.

**Comment by feerrenrut** [comment]
@kvark128 There are some outstanding questions from josephsl that we would like to see answered before we re-review the code. On the other hand, there has been no movement on this pull request for a long time, so if someone else is interested in this feature, please feel free to take it on.

**Comment by josephsl** [comment]
Hi,

I have practical experience with coding search history (see my SPL add-on for details), so I'd be happy to take this on or mentor someone. However, to make it easy for folks to review that feature, I recommend not working on this until #7104 is a reality within master branch. Also, a new PR should be created with an actual accompanying issue because we need to gather more cases for search history.

For now, I'll take this offline unless someone wants this to be reopened but on fulfilling the following conditions: that outstanding review items are addressed along with other possible concerns from users, commits coming from a topic branch (branch other than one's own master branch, and the work is based on wxPython 4 (because of GUI changes).

Thanks.

**Pull request closed without merging**